0000-rep-template.md

```diff
@@ -0,0 +1,45 @@
+---
+REP Number: <Fill me in with a four-digit unique incrementing number, matching the pull request number; update AFTER REP acceptance.>
+Author: <Author's First Name and Last Name.>
+Start Date: <Today's Date, YYYY-MM-DD>
+Bicep Issue Number: <Fill if the REP originates from a Bicep issue; otherwise, delete.>
+Enhances: <Fill an existing final REP number if this REP enhances it; otherwise, delete.>
+Deprecates: <Fill an existing final REP number if this REP deprecates it; otherwise, delete.>
+Enhanced By: <Applicable to existing final REPs only. Fill a final REP number if that REP enhances this one. Delete for new REPs.>
+Deprecated By: <Applicable to existing final REPs only. Fill a final REP number if that REP deprecates this one. Delete for new REPs.>
+---
+
+# Title - replace me with feature name
+
+## Summary
+
+Provide a brief paragraph explaining the feature.
+
+## Motivation
+
+Outline the reasons behind implementing this feature, the use cases it supports, and the expected outcomes.
+
+## Detailed design
+
+The core of the REP. Elaborate on the design and implementation with sufficient detail for someone familiar with Bicep to grasp. Provide detailed examples to illustrate how the feature is used and its implications on user experience.
+
+### Examples
+
+- Example 1
+```bicep
+```
+- Example 2
+```bicep
+```
+
+## Drawbacks
+
+Discuss any potential drawbacks associated with this approach. Why might this *not* be a suitable choice? Consider implementation costs, integration with existing and planned features, and the migration cost for Bicep users. Clearly identify if it constitutes a breaking change.
+
```

Comment:
Should we add an Out of Scope:
> [Out of scope](https://gist.github.com/rynowak/5a39b92997546216f957f0aa2dc4b32b#out-of-scope)
Describe the important issues or concerns NOT addressed by this proposal. 1 paragraph each (can be short)

Often the "out of scope" items are refer to the current state. We might already have planned work to address the "out of scope" items, mention it if that's the case".

Reply:
Added an Out of scope section.